### PR TITLE
Migrate library to .NET Standard

### PR DESCRIPTION
**Pull request by Jimmyson**
Looking to utilize the library in a .NET Core application, and thought it would be useful to have this migrated, supported and published to NuGet.

Namespaces have been preserved, and dependent libraries have been updated.
Unit tests still pass.
Removed Example WinForms application as no code remained.

### CODE REVIEW

**Comment by wo80** [comment]
Thanks, I fixed the appveyor test runner.

Two remarks:

You made a lot of previously internal classes public. Is this because of the tests or is there any other reason? If it's because of the tests, you could add
```c#
[assembly: InternalsVisibleTo("AcoustID.Tests")] // or AcoustID.Core.Tests
```
to one of the source files and revert the changes (public -> internal).

Additionally, I don't like the `Core` suffix too much. Any good reason to add it to the project file names?

File: src/AcoustID.Core/AcoustID.Core.csproj [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
```
Comment (wo80) [inline src/AcoustID.Core/AcoustID.Core.csproj]: Please add a property group containing the assembly information:
```xml
  <PropertyGroup>
    <GeneratePackageOnBuild>false</GeneratePackageOnBuild>
    <GenerateDocumentationFile>true</GenerateDocumentationFile>
    <PackageId>AcoustID.NET</PackageId>
    <Description>AcoustID fingerprinter and webservice access.</Description>
    <Product>AcoustID</Product>
    <Copyright>Copyright © Christian Woltering 2016-2019</Copyright>
    <Authors>Christian Woltering</Authors>
    <PackageProjectUrl>https://github.com/wo80/AcoustID.NET</PackageProjectUrl>
    <RepositoryUrl>https://github.com/wo80/AcoustID.NET</RepositoryUrl>
    <AssemblyVersion>1.3.3.0</AssemblyVersion>
    <FileVersion>1.3.3.0</FileVersion>
    <PackageTags>music audio fingerprint acoustid chromaprint</PackageTags>
    <Version>1.3.3</Version>
    <AssemblyName>AcoustID</AssemblyName>
    <RootNamespace>AcoustID</RootNamespace>
    <Configurations>Debug;Release</Configurations>
    <PackageLicenseUrl>https://github.com/acoustid/chromaprint/blob/master/LICENSE.md</PackageLicenseUrl>
  </PropertyGroup>
```

**Review by wo80 — CHANGES_REQUESTED** [review]

**Comment by Jimmyson** [comment]
My reason for adding Core was to indicate the change to Core framework, it was only in name though and I can revert this.

In regards to opening up the internal classes, I had access issues. I did this some time ago, and did not know about the assembly directive.

I can reattempt this to keep in line with access structure...

**Comment by wo80** [comment]
If you need access to those classes for your work, I don't mind making them public. I was just wondering ...

But I'd prefer if you remove `Core` from directory and project file names. Thanks!

**Comment by Jimmyson** [comment]
I have taken a branch of the original fork, and followed through with checking the to see if's possible to expose the Internal classes. And the tests successfully pass.

image

There is an attribute that can be added to the primary library's CSPROJ file that allows the "InternalVisableTo" flag
```xml
<ItemGroup>
    <AssemblyAttribute Include="System.Runtime.CompilerServices.InternalsVisibleTo">
        <_Parameter1>$(AssemblyName).Tests</_Parameter1>
    </AssemblyAttribute>
</ItemGroup>
```
https://blog.sanderaernouts.com/make-internals-visible-with-new-csproj-format

I can revert the file back to their original state, but since I changed the folder name to include the `Core` value. I will need to rollback the branch. Apologies for that.

**Comment by Jimmyson** [comment]
I have rolled back my changes and reattempted the migration.
Preserving the Internals and removing the `Core` name from folders too.
I wasn't too sure about if you were also concerned about the `test` and `src` folders, so reverted that too.

But I have now implemented the defined changes you have requested for `Internals` and the property group

File: src/AcoustID.Core/AcoustID.Core.csproj (hunk 1 above)
Comment (Jimmyson) [reply src/AcoustID.Core/AcoustID.Core.csproj]: Changed have been implemented after the revert. See `AcoustID.csproj` in 085088c

**Review by wo80 — APPROVED** [review]

**Comment by wo80** [comment]
Ok, this looks much cleaner. I'll merge and publish a new package this weekend.

**Pull request merged**